### PR TITLE
Use winston logger and log to Stackdriver Logging

### PR DESCRIPTION
**Pull request by bonomat**
Unfortunately log4js did not support stackdriver logging out of the box, hence I went for the recommended solution from google: [winston](https://github.com/winstonjs/winston)

* log.trace does not exist, hence I added a new log level so that we don't have to change our log logic
* it can log to different transport layers, e.g. gcloud and console
* it supports different format per transport layers

Console logging: 
image

GCloud logging: 
image

Note: we should disable console logging on our VM so that logs don't show up twice.

### CODE REVIEW

File: src/actionExecutor.ts [hunk 1]
```diff
@@ -39,7 +39,10 @@ export class ActionExecutor {
     let result: Result<any, Error>;
 
     const triggerResult = await this.triggerRequestFromAction(action);
-    logger.trace(`Response from action: ${JSON.stringify(triggerResult)}`);
+    logger.log(
+      "trace",
+      `Response from action: ${JSON.stringify(triggerResult)}`
```
Comment (D4nte) [inline src/actionExecutor.ts]:
```suggestion
      "Response from action:",
       triggerResult`
```

We discussed that in the past, that we should pass the object directly to allow stackdriver to actually recognize the object and allow us to do research on such object properties. Is that still valid?

See the following quote in the original issue:
> log4js is more powerful than debug and supports arbitrary parameters to its log-fn calls: https://github.com/log4js-node/log4js-node/blob/master/examples/example.js#L38

File: src/rates/staticRates.ts [hunk 2]
```diff
@@ -25,7 +25,9 @@ export default class StaticRates implements TradeService {
 
     if (!rate) {
       logger.warn(
-        `Rate not configured for buy: ${buy.asset}, sell: ${sell.asset}`
+        `Rate not configured for buy and sell: `,
```
Comment (da-kami) [inline src/rates/staticRates.ts]:
```suggestion
        "Rate not configured for buy and sell: ",
```

File: src/actionExecutor.ts [hunk 3]
```diff
@@ -39,14 +39,14 @@ export class ActionExecutor {
     let result: Result<any, Error>;
 
     const triggerResult = await this.triggerRequestFromAction(action);
-    logger.trace(`Response from action: ${JSON.stringify(triggerResult)}`);
+    logger.log("trace", `Response from action`, triggerResult);
```
Comment (tcharding) [inline src/actionExecutor.ts]: Can we implement a helper function that wraps `logger.log("trace", "awesome loge message")` for this to reduce the number of characters (assuming trace level logging is a common use case).  Something like `traceLog("even awesomer log message")`?

File: src/bitcoin/bitcoinCoreRpc.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ export class BitcoinCoreRpc implements BitcoinBlockchain {
       scanobjects
     );
     if (!result || !result.success || result.unspents === undefined) {
-      logger.error(`Transaction scan failed: ${result}`);
+      logger.crit(`Transaction scan failed: `, result);
```
Comment (tcharding) [inline src/bitcoin/bitcoinCoreRpc.ts]: This change and the one above are not uniform.  One uses `%` for string substitution and one does not.  Is there a reason for this?

**Pull request merged**